invenio_communities/assets/semantic-ui/js/invenio_communities/settings/curation-policy.js

```diff
@@ -0,0 +1,133 @@
+      metadata: {
+        curation_policy: "",
+      },
+    });
+
+    return initialValues;
+  };
+
+  setGlobalError = (errorMsg) => {
+    this.setState({ error: errorMsg });
+  };
+
+  onSubmit = async (values, { setSubmitting, setFieldError }) => {
+    const { community } = this.props;
+    setSubmitting(true);
+
+    try {
+      const client = new CommunityApi();
+      await client.update(community.id, values);
+      window.location.reload();
+    } catch (error) {
+      if (error === "UNMOUNTED") return;
+
+      const { message, errors } = communityErrorSerializer(error);
+
+      if (message) {
+        this.setGlobalError(message);
+      }
+
+      if (errors) {
+        errors.forEach(({ field, messages }) => setFieldError(field, messages[0]));
+      }
+    }
+
+    setSubmitting(false);
+  };
+  render() {
+    const { error } = this.state;
+    const { community } = this.props;
+    return (
+      <Formik
+        initialValues={this.getInitialValues(community)}
+        validationSchema={COMMUNITY_VALIDATION_SCHEMA}
+        onSubmit={this.onSubmit}
+      >
+        {({ isSubmitting, isValid, handleSubmit }) => (
+          <Form onSubmit={handleSubmit}>
+            <Message hidden={error === ""} negative className="flashed">
+              <Grid container>
+                <Grid.Column width={15} textAlign="left">
+                  <strong>{error}</strong>
+                </Grid.Column>
+              </Grid>
+            </Message>
+            <Grid>
+              <Grid.Row className="pt-10 pb-0">
+                <Grid.Column mobile={16} tablet={16} computer={12} className="rel-pb-2">
+                  <RichInputField
+                    fieldPath="metadata.curation_policy"
+                    label={
+                      <FieldLabel
+                        htmlFor="metadata.curation_policy"
+                        icon="pencil"
+                        label={i18next.t("Curation policy")}
+                      />
+                    }
+                    editorConfig={CKEditorConfig}
+                    fluid
+                  />
+                  <Button
+                    primary
+                    icon
+                    labelPosition="left"
+                    loading={isSubmitting}
+                    disabled={!isValid || isSubmitting}
+                    toggle
+                    type="submit"
+                  >
+                    <Icon name="save" />
+                    {i18next.t("Save")}
+                  </Button>
+                </Grid.Column>
+              </Grid.Row>
+            </Grid>
+          </Form>
+        )}
+      </Formik>
+    );
+  }
+}
+
+CommunityCurationPolicyForm.propTypes = {
+  community: PropTypes.object.isRequired,
+};
+
+const domContainer = document.getElementById("app");
+const community = JSON.parse(domContainer.dataset.community);
+
+ReactDOM.render(<CommunityCurationPolicyForm community={community} />, domContainer);
```

Comment:
could this be moved to separate `index.js` file ? then we keep only the component class here